### PR TITLE
feat(pre-post-request-scripts): use `postman-sandbox`

### PR DESCRIPTION
**Pull request by hanspagel**
## Problem

* so many people have Postman scripts, and they want to execute them in Scalar
* but we only mocked the `pm` JS api partially, including some `chai` assertions
* that's incomplete and a lot of scripts will fail

```js
pm.test("Status code is 200", () => {
  pm.response.to.have.status(200)
})
```

## Solution

this PR switches to `postman-sandbox` to have 1:1 feature parity with their API

## Checklist

- [x] I explained why the change is needed.
- [ ] I added a changeset. 
- [x] I added tests.
- [ ] I updated the documentation.

---

> [!NOTE]
> **Medium Risk**
> Changes the core script execution runtime and how responses/assertions are interpreted, which may alter behavior for existing scripts and has some risk of edge-case incompatibilities despite added tests.
> 
> **Overview**
> Post-response scripts in `@scalar/pre-post-request-scripts` now execute inside the official `postman-sandbox` runtime instead of the repo’s custom `pm`/assertion implementation, aiming for Postman-compatible semantics and behavior.
> 
> This introduces a `postman-sandbox` adapter that converts a Fetch `Response` into the sandbox’s expected shape, forwards sandbox console output to the plugin console, and maps sandbox assertion events into `TestResult` updates (including per-assertion durations) with guaranteed listener/context cleanup.
> 
> Removes the previously exported internal `postman-scripts` context utilities and synchronous response/types, updates docs and build/tsconfig accordingly, and adds focused tests for sandbox execution, header handling, and runtime error reporting (plus a changeset bump).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit fff21036ede9dec1a9b23ad7dbf199ef5e0f9aa0. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/pre-post-request-scripts/src/libs/execute-scripts/postman-sandbox-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            onTestResultsUpdate?.([...testResults])
+          }
+
+          resolve()
+        },
+      )
+    })
+  } finally {
+    sandboxContext.off('execution.assertion', handleAssertion)
+    sandboxContext.off('console', handleConsole)
```
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/pre-post-request-scripts/src/libs/execute-scripts/postman-sandbox-adapter.ts:146]: need to fix that, converted to draft
maybe need to revert the previous commit

**Comment by amritk** [comment]
can we also get typing/autocomplete in the scripts code snippet

**Comment by hanspagel** [comment]
@amritk yes, but in a separate PR pls :))

File: packages/pre-post-request-scripts/src/libs/execute-scripts/postman-sandbox-adapter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/pre-post-request-scripts/src/libs/execute-scripts/postman-sandbox-adapter.ts:146]: reverted, is easier
